### DISCUSSION TITLE
Re-fitting BVH for modified scene

### DISCUSSION THREAD

**Discussion by vani-nag** [owl-project/owl#164] [Q&A] [answered]
I'm trying to refit my BVH of spheres to account for modified radii and centers. I'm running into an issue, where the modified radii and centers are visible in OPTIX_BOUNDS_PROGRAM, but not in OPTIX_INTERSECT_PROGRAM. These are the steps I followed to refit the bvh:
1) Similar to samples/cmdline/s05-rtow, the original BVH build:
  ```
OWLBuffer SpheresBuffer
    = owlDeviceBufferCreate(context,OWL_USER_TYPE(Spheres[0]),
                            Spheres.size(),Spheres.data());
                            
  OWLGeom SpheresGeom
    = owlGeomCreate(context,SpheresGeomType);
  owlGeomSetPrimCount(SpheresGeom,Spheres.size());
  owlGeomSetBuffer(SpheresGeom,"prims",SpheresBuffer);
  owlGeomSet1f(SpheresGeom,"rad",radius);

// ##################################################################
// set up all *ACCELS* we need to trace into those groups
// ##################################################################

  OWLGroup spheresGroup
  = owlUserGeomGroupCreate(context,1,userGeoms);
owlGroupBuildAccel(spheresGroup);

OWLGroup world
  = owlInstanceGroupCreate(context,1,&spheresGroup);
owlGroupBuildAccel(world);

owlLaunch2D(rayGen,...)
```
2) I increment radius+= 0.5 and refit as follows:
```
//Update radius
owlGeomSet1f(SpheresGeom,"rad",radius);

// I update the Spheres vector to modify the centers
SpheresBuffer = owlDeviceBufferCreate(context,OWL_USER_TYPE(Spheres[0]), Spheres.size(),Spheres.data());
owlGeomSetBuffer(SpheresGeom,"prims",SpheresBuffer);
 
// ##################################################################
// set up all *ACCELS* we need to trace into those groups
// ##################################################################

owlGroupRefitAccel(spheresGroup);
owlGroupRefitAccel(world);

owlLaunch2D(rayGen,...)
  ```
Please let me know if I'm missing something. Thanks!

**Comment by ingowald** [comment, marked as answer]
I assume you have forgotten/omitted a "owlBuildSBT" call after changing the radius: For any of the "default" RTX programs (intersect, closesthit, etc) the parameters are pulled from the SBT, so this needs rebuilding if paramters change. The reason that it *does* work for the bounds program is that we handle the bounds prog outside the RTX pipeline, so the parameters are pulled from another place that does not require rebuilding the SBT.
